### PR TITLE
feat(message-queue): create message-queue package

### PR DESCRIPTION
**Pull request by jakim929**
thin wrapper around bullmq that 
- enforces namespaces
- adds basic logging
- adds better typing job job name/data
- adds JobProcessor - a convenience class that lets you set up handlers easily / typesafe way
- see `/examples` for usage examples

### CODE REVIEW

**Comment by jakim929** [comment]
* **#364** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/ecosystem/364?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @jakim929 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by tremarkley — APPROVED** [review]
great job!

File: packages/message-queue/src/examples/example-queue.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import type { QueueConfig } from '@/config/QueueConfig'
+import type { ExampleQueueJobParams } from '@/examples/example-config'
+import { createQueue } from '@/queue/createQueue'
+
+const queueConfig = {
+  namespace: 'gateway',
+  queueName: 'example-queue',
+  redisUrl: 'redis://localhost:6379',
+} as const satisfies QueueConfig
```
[the remark this answers is not in the document]
Comment (jakim929) [reply packages/message-queue/src/examples/example-queue.ts:9]: yep! will fix

File: packages/message-queue/src/job-processor/JobProcessor.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import type { QueueJobParams } from '@/config/QueueJobParams'
+
+export class JobProcessor<TQueueConfig extends QueueJobParams> {
+  private handlerByName: Record<
+    keyof TQueueConfig,
+    (jobData: any) => Promise<void>
```
[the remark this answers is not in the document]
Comment (jakim929) [reply packages/message-queue/src/job-processor/JobProcessor.ts:6]: will update types!

File: packages/message-queue/src/job-processor/JobProcessor.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+  // register a handler to a job name
+  handle<TJobName extends keyof TQueueConfig>(
+    jobName: TJobName,
+    handler: (jobData: TQueueConfig[TJobName]) => Promise<void>,
+  ) {
+    this.handlerByName[jobName] = handler
+  }
+
+  async process(job: { name: string; data: any }) {
```
[the remark this answers is not in the document]
Comment (jakim929) [reply packages/message-queue/src/job-processor/JobProcessor.ts:17]: this one if intentional - the process function should handle case when a non-supported job / data type is passed in.

**Pull request merged**